phoenix-core/src/main/java/org/apache/phoenix/iterate/TableSnapshotResultIterator.java

```diff
@@ -65,7 +66,8 @@ public TableSnapshotResultIterator(Configuration configuration, Scan scan, ScanM
     this.scan = scan;
     this.scanMetricsHolder = scanMetricsHolder;
     this.scanIterator = UNINITIALIZED_SCANNER;
-    this.restoreDir = new Path(configuration.get(PhoenixConfigurationUtil.RESTORE_DIR_KEY));
+    this.restoreDir = new Path(new Path(configuration.get(PhoenixConfigurationUtil.RESTORE_DIR_KEY)),
```

Comment:
`Path` also offers other API at https://hadoop.apache.org/docs/r2.8.2/api/org/apache/hadoop/fs/Path.html
You can specify both of the params as `String`